pineappl/src/evolution.rs

```diff
@@ -499,10 +499,10 @@ pub(crate) fn evolve_slice_with_one(
         }));
     }
 
-    let pid = if has_pdf1 {
-        grid.initial_state_2()
+    let pid = if grid.convolutions()[0] != Convolution::None {
+        grid.lumi()[0].entry()[0].1
```

Comment:
this seems wrong to me: lumi and convolutions are playing on two different levels - maybe you meant `grid.convolutions()[1].pid()` here? (same below) (this has the usual caveat mentioned before, that `[1]` has to exist)